uisdk/src/main/java/com/karhoo/uisdk/screen/booking/quotes/filterview/FilterDialogFragment.kt

```diff
@@ -0,0 +1,145 @@
+            dismiss()
+        }
+
+        filterViewResetFilters = view.findViewById(R.id.filterViewResetFilters)
+        filterViewResetFilters.setOnClickListener {
+            presenter.resetFilters()
+            dismiss()
+        }
+
+        filterViewPassengerNumberedFilter =
+            view.findViewById<NumberedFilterView>(R.id.filterViewPassengerNumberedFilter)
+        filterViewLuggageNumberedFilter =
+            view.findViewById<NumberedFilterView>(R.id.filterViewLuggageNumberedFilter)
+        filterViewVehicleTypeMultiSelectChipsFilter =
+            view.findViewById<MultiSelectChipsFilterView>(R.id.filterViewVehicleTypeMultiSelectChipsFilter)
+
+        presenter.createFilters()
+        return view
+    }
+
+    fun setListener(filterDelegate: FilterDialogPresenter.FilterDelegate) {
+        presenter.setFilterDelegate(filterDelegate)
+    }
+
+    fun createFilterChain(filterChain: FilterChain) {
+        presenter.filterChain = filterChain
+
+        passengersFilter = PassengersFilter(1)
+        luggageFilter = LuggageFilter(0)
+
+        filterChain.filters.add(passengersFilter)
+        filterChain.filters.add(luggageFilter)
+    }
+
+    companion object {
+        const val TAG = "FilterView"
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = BottomSheetDialog(requireContext(), theme)
+        dialog.setOnShowListener {
+
+            val bottomSheetDialog = it as BottomSheetDialog
+            val parentLayout =
+                bottomSheetDialog.findViewById<View>(com.google.android.material.R.id.design_bottom_sheet)
+            parentLayout?.let { it ->
+                val behaviour = BottomSheetBehavior.from(it)
+                setupFullHeight(it)
+                behaviour.state = BottomSheetBehavior.STATE_EXPANDED
+            }
+        }
+        return dialog
+    }
+
+    private fun setupFullHeight(bottomSheet: View) {
+        val layoutParams = bottomSheet.layoutParams
+        layoutParams.height = WindowManager.LayoutParams.MATCH_PARENT
+        bottomSheet.layoutParams = layoutParams
+    }
+
+    override fun createFilters(filterChain: FilterChain) {
+        filterViewPassengerNumberedFilter.filter = passengersFilter
+        filterViewPassengerNumberedFilter.setTitle(getString(R.string.kh_uisdk_filter_passengers))
+        filterViewPassengerNumberedFilter.icon = R.drawable.kh_uisdk_ic_passengers
+        filterViewPassengerNumberedFilter.delegate = {
+            presenter.callFilterChanged()
+        }
+
+        filterViewLuggageNumberedFilter.filter = luggageFilter
+        filterViewLuggageNumberedFilter.setTitle(getString(R.string.kh_uisdk_filter_luggages))
+        filterViewLuggageNumberedFilter.icon = R.drawable.kh_uisdk_ic_luggage
+        filterViewLuggageNumberedFilter.delegate = {
+            presenter.callFilterChanged()
+        }
+//
+//        val vehicleTypeFilter = VehicleTypeFilter(ArrayList()).apply {
```

Comment:
Could this chunk be cleaned up?

Reply:
I left them there because I already did them and in the next ticket will be needed there. I can clean them up but I may need to add them soon.